src/datasources/data-decoder-api/data-decoder-api.manager.ts

```diff
@@ -0,0 +1,51 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { DataDecoderApi } from '@/datasources/data-decoder-api/data-decoder-api.service';
+import { HttpErrorFactory } from '@/datasources/errors/http-error-factory';
+import {
+  NetworkService,
+  INetworkService,
+} from '@/datasources/network/network.service.interface';
+import { IDataDecoderApiManager } from '@/domain/interfaces/data-decoder-api.manager.interface';
+
+@Injectable()
+export class DataDecoderApiManager implements IDataDecoderApiManager {
```

Comment:
While I think it's nice to reuse the "manager" pattern we have in other places, I'm not 100% sure it fits this case, as the datasource is chain-agnostic. It will work anyway, and it's solid from an architectural standpoint, but I also think it hides some capabilities of the datasource, like getting all the contracts for several chains at once (as each `DataDecoderApi` is limited to `this.chainId`). Not a strong opinion or a request for change, but I wanted to know your thoughts on this :)

Reply:
I agree. I removed it in 01ea3dcd9a518d3da2cec7409906d9a758b3ac52.